crates/astria-sequencer/src/assets/state_ext.rs

```diff
@@ -20,46 +25,23 @@ use tendermint::abci::{
 };
 use tracing::instrument;
 
-use super::storage;
+use super::storage::{
+    self,
+    keys::{
+        asset_key,
+        block_fees_key,
+        extract_asset_from_block_fees_key,
+        extract_asset_from_fee_asset_key,
+        fee_asset_key,
+        BLOCK_FEES_PREFIX,
+        FEE_ASSET_PREFIX,
+        NATIVE_ASSET_KEY,
+    },
+};
 use crate::storage::StoredValue;
 
-const BLOCK_FEES_PREFIX: &str = "block_fees/";
-const FEE_ASSET_PREFIX: &str = "fee_asset/";
-const NATIVE_ASSET_KEY: &str = "nativeasset";
-
-fn asset_storage_key<'a, TAsset>(asset: &'a TAsset) -> String
-where
-    asset::IbcPrefixed: From<&'a TAsset>,
-{
-    format!("asset/{}", crate::storage_keys::hunks::Asset::from(asset))
-}
-
-fn block_fees_key<'a, TAsset>(asset: &'a TAsset) -> String
-where
-    asset::IbcPrefixed: From<&'a TAsset>,
-{
-    format!(
-        "{BLOCK_FEES_PREFIX}{}",
-        crate::storage_keys::hunks::Asset::from(asset)
-    )
-}
-
-fn fee_asset_key<'a, TAsset>(asset: &'a TAsset) -> String
-where
-    asset::IbcPrefixed: From<&'a TAsset>,
-{
-    format!(
-        "{FEE_ASSET_PREFIX}{}",
-        crate::storage_keys::hunks::Asset::from(asset)
-    )
-}
-
 /// Creates `abci::Event` of kind `tx.fees` for sequencer fee reporting
-fn construct_tx_fee_event<T: std::fmt::Display>(
-    asset: &T,
-    fee_amount: u128,
-    action_type: String,
-) -> Event {
+fn construct_tx_fee_event<T: Display>(asset: &T, fee_amount: u128, action_type: String) -> Event {
```

Comment:
While we are at it, can we could make this `fn construct_tx_fee_event<T: Display, U: Protobuf>(asset: &T, fee_amount: u128) -> Event {` and then in the function body use `<U as Protobuf>::full_name`

Reply:
Done in [bb0c659](https://github.com/astriaorg/astria/pull/1616/commits/bb0c65935434d38b7028291c2ce4b774bfe7d78b).